ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContext.java

```diff
@@ -162,6 +162,16 @@ protected void create() {
 		fDocumentProvider = createDocumentProvider(fEditorInput);
 		try {
 			fDocumentProvider.connect(fEditorInput);
+			// Verify that the document is available after connect()
+			// This can fail if the file was deleted/moved during editor initialization
+			// or if there's a race condition with workspace events
+			IDocument document = fDocumentProvider.getDocument(fEditorInput);
+			if (document == null) {
+				throw new CoreException(new org.eclipse.core.runtime.Status(
+					org.eclipse.core.runtime.IStatus.ERROR,
+					"org.eclipse.pde.ui", //$NON-NLS-1$
+					"Document not available after connecting to document provider for input: " + fEditorInput.getName())); //$NON-NLS-1$
```

Comment:
Does this substantially improve the situation? To me it seems like we are not trading a `NullPointerException` for a a `CoreException`?

Reply:
Furthermore calling the `new Status` constructor can be replaced by the simpler `Status.error()` factory:
```suggestion
				throw new CoreException(Status.error("Document not available after connecting to document provider for input: " + fEditorInput.getName())); //$NON-NLS-1$
```